lib/src/components/NxSegmentedButton/__tests__/NxSegmentedButton.test.tsx

```diff
@@ -76,47 +80,47 @@ describe('NxSegmentedButton', function() {
   });
 
   it('renders nx-segmented-btn__main-btn and nx-segmented-btn__dropdown-btn NxButtons as children', function() {
-    expect(getShallow()).toContainMatchingElement('.nx-segmented-btn__main-btn');
-    expect(getShallow()).toContainMatchingElement('.nx-segmented-btn__dropdown-btn');
-
-    expect(getShallow().find('.nx-segmented-btn__main-btn')).toMatchSelector(NxButton);
-    expect(getShallow().find('.nx-segmented-btn__dropdown-btn')).toMatchSelector(NxButton);
+    expect(getMounted()).toContainMatchingElement('.nx-segmented-btn__main-btn');
+    expect(getMounted()).toContainMatchingElement('.nx-segmented-btn__dropdown-btn');
+    const mainButton = getMounted().find('.nx-segmented-btn__main-btn').at(0);
+    expect(mainButton).toMatchSelector(NxButton);
+    expect(getMounted().find('.nx-segmented-btn__dropdown-btn').at(0)).toMatchSelector(NxButton);
   });
 
   it('sets the variant on the child buttons', function() {
-    const primaryComponent = getShallow(),
-        secondaryComponent = getShallow({ variant: 'secondary' });
+    const primaryComponent = getMounted(),
+        secondaryComponent = getMounted({ variant: 'secondary' });
 
-    expect(primaryComponent.find('.nx-segmented-btn__main-btn')).toHaveProp('variant', 'primary');
-    expect(primaryComponent.find('.nx-segmented-btn__dropdown-btn')).toHaveProp('variant', 'primary');
+    expect(primaryComponent.find('.nx-segmented-btn__main-btn').at(0)).toHaveProp('variant', 'primary');
```

Comment:
You can include react component names in string selectors here as if they were html tag names.  That would probably be cleaner.

Reply:
> You can include react component names in string selectors here as if they were html tag names. That would probably be cleaner.

Hmm I can't seem to get the selector to work `NxButton.nx-segmented-btn__main-btn` (without the `.at(0)`).
It returns an empty node for me.

Reply:
> Hmm I can't seem to get the selector to work NxButton.nx-segmented-btn__main-btn (without the .at(0)).
It returns an empty node for me.

Oh it's because of forwardRef ugliness. It seems to work if you write it as `ForwardRef(NxButton).nx-segmented-btn__main-btn`

Btw for the future you can add lines like `console.log(primaryComponent.debug());` to diagnose this sort of thing

Reply:
> > Hmm I can't seem to get the selector to work NxButton.nx-segmented-btn__main-btn (without the .at(0)).
> > It returns an empty node for me.
> 
> Oh it's because of forwardRef ugliness. It seems to work if you write it as `ForwardRef(NxButton).nx-segmented-btn__main-btn`
> 
> Btw for the future you can add lines like `console.log(primaryComponent.debug());` to diagnose this sort of thing

Thanks so much, I didn't know you can write it like `ForwardRef(NxButton)`! and the `.debug()` that's very helpful.
Fixed in this commit: 0242c3a2dac0d099ed8fe2a9d1f51f89a6d4633f